Framework/Algorithms/src/CalculateMultipleScattering.cpp

```diff
@@ -311,7 +311,9 @@ void CalculateMultipleScattering::exec() {
   PARALLEL_CHECK_INTERUPT_REGION
 
   if (useSparseInstrument) {
-    const std::string reportMsg = "Interpolating";
+    Poco::Thread::sleep(200); // to ensure prog message changes
```

Comment:
We've not had to do this before when emitting progress messages. I assume just nothing changed without this?

Reply:
The following code is the reason this sleep was necessary. It causes there to be a min of 0.1 seconds between progress updates so if you issue two in quick succession where the "stem" of the message has changed (in my case from "Computing corrections" to "Spatial interpolation"), the 2nd message doesn't get displayed:

```
void AlgorithmProgressPresenter::updateProgressBarSlot(Mantid::API::AlgorithmID algorithm, const double progress,
                                                       const QString message, const double estimatedTime,
                                                       const int progressPrecision) {
  if (algorithm == this->m_algorithm) {
    // this needs to be a call to the view
    // so that it can be mocked out for testing
    constexpr float maxRefreshInterval{0.1f};
    float timeInterval = m_timer.elapsed_no_reset();
    if (timeInterval > maxRefreshInterval) {
      m_timer.reset();
      m_view->updateProgress(progress, message, estimatedTime, progressPrecision);
    }
  }
}
```
Perhaps ideally the maxRefreshInterval should be called minRefreshInterval. Also it seems like the timeInterval should only be checked if the message is the same as last time. I think by this time "message" has had the progress % appended to it though so hard to tell if the stem of the message has changed.

Reply:
Oh yeah I see. I think we put a limit on this as there were cases where too many reports were going out in quick succession and slowing the GUI down. I agree that this is minimum time though and not a maximum and that the message should be checked but that is definitely outside of the scope of this issue.